### DISCUSSION TITLE
[Arma 3] Mixed use of path separators when loading mods

### DISCUSSION THREAD

**Discussion by palten08** [GameServerManagers/LinuxGSM#3947] [Q&A]
Server OS: Ubuntu 22.04.1 LTS

I can't tell this if is a problem with the Arma 3 server binary or LinuxGSM (I have a feeling it's the former), but I've noticed in the logs that none of my mods are loading, and from what I can gather I think it's because there's a mixture between back and forward-slash path separators being used in the path:

> 23:44:06 SteamAPI initialization failed. Steam features won't be accessible!
23:44:06 Initializing stats manager.
sh: 1: /sbin/ifconfig: not found
23:44:06 Stats config disabled.
23:44:06 sessionID: f1557a591f1792ceaa0d6963b03841c521bcb778
ErrorMessage: Cannot open file '/data/serverfiles/mods/@xyictsfo\addons\xyi_ctsfo_mw'
Application terminated intentionally
23:44:06 ../lib/Network/networkServer.cpp ClearNetServer:NOT IMPLEMENTED - briefing!
Cannot open file '/data/serverfiles/mods/@xyictsfo\addons\xyi_ctsfo_mw'
23:44:06 Critical:Destroying running thread!
23:44:06 Extensions:

I also can't seem to shake the feeling that only the last mod in the mods array string is the one getting loaded. Here's the string for reference:

`mods="mods/@alphagroupequipment\;mods/@asip\;mods/@aveightbharriertwo\;mods/@blkout\;mods/@cbaa3\;mods/@cupterrainscore\;mods/@cupterrainsmaps\;mods/@cupterrainsmaps20\;mods/@cupunits\;mods/@cupvehicles\;mods/@cupweapons\;mods/@edenobjects\;mods/@f15eagle\;mods/@f16fightingfalcon\;mods/@fa18hornet\;mods/@firawsairweaponsystem\;mods/@gearcollectiondelta\;mods/@ilyushinil76\;mods/@lambsdangerfsm\;mods/@mich20002001basepack\;mods/@moepilotgearsuite\;mods/@mwsas\;mods/@niarmsaugrifles\;mods/@niarmscore\;mods/@niarmsfalrifles\;mods/@niarmsg3rifles\;mods/@niarmsmp5smgs\;mods/@powerrangers\;mods/@projectopfor\;mods/@rhsafrf\;mods/@rhsafrfdev\;mods/@rhsgref\;mods/@rhsgrefdev\;mods/@rhssaf\;mods/@rhssafdev\;mods/@rhsusaf\;mods/@rhsusafdev\;mods/@seventyfifth\;mods/@su35flankere\;mods/@superpowerrangers\;mods/@superpowerusmc\;mods/@tfl\;mods/@thesukhoisu33flankerd\;mods/@tieroneweapons\;mods/@usafmodfighters\;mods/@usafmodmain\;mods/@usafmodutility\;mods/@xyicore\;mods/@xyictsfo\;"`

Am I doing something wrong? I'm following every example I could find to the T, but I haven't seen anybody discussing an issue where two different path separators are being used, so I have no idea how to get around that if I'm honest.

**Comment by peejaybee** [comment]
I can't say why you are getting that error about not finding files, but I had an issue just today where only the last mod in the list would actually load. I tracked it down to the mod list being wrapped in quotes in the -mod param that is generated from the mod list in the config.  I will be entering an issue and a PR to fix this.
